src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CppCodegenNodeFactory.cs

```diff
@@ -38,5 +44,33 @@ protected override ISymbolNode CreateReadyToRunHelperNode(Tuple<ReadyToRunHelper
             // TODO: this is wrong: this returns an assembly stub node
             return new ReadyToRunHelperNode(helperCall.Item1, helperCall.Item2);
         }
+
+        private void AddWellKnownType(WellKnownType wellKnownType, DependencyAnalysisFramework.DependencyAnalyzerBase<NodeFactory> graph)
+        {
+            var type = TypeSystemContext.GetWellKnownType(wellKnownType);
+            var typeNode = ConstructedTypeSymbol(type);
+            graph.AddRoot(typeNode, "Enables CPP codegen");
+        }
+        private void AddWellKnownTypes(DependencyAnalysisFramework.DependencyAnalyzerBase<NodeFactory> graph)
+        {
+
+            AddWellKnownType(WellKnownType.Void, graph);
```

Comment:
Did you look into finding a way to not duplicate the list from `_wellKnownTypeNodes` from `CppWriter` in here? I dislike that we're setting a precedent here to root a bunch of things without explanation.

(I assume `_wellKnownTypeNodes` is the reason why we need this rooting in the first place.)